src/dbnode/storage/bootstrap/bootstrapper/commitlog/source.go

```diff
@@ -97,169 +135,335 @@ func newCommitLogSource(opts Options, inspection fs.Inspection) bootstrap.Source
 	)
-
+	s.log.Info("read commit logs start")
 	defer func() {
-		s.log.Info("ReadData finished",
-			zap.Int("seriesSkipped", seriesSkipped),
+		datapointsSkipped := 0
+		datapointsRead := 0
+		for _, worker := range workers {
+			datapointsSkipped += worker.datapointsSkipped
+			datapointsRead += worker.datapointsRead
+		}
+		s.log.Info("read finished",
+			zap.Stringer("took", s.nowFn().Sub(startCommitLogsRead)),
 			zap.Int("datapointsSkipped", datapointsSkipped),
 			zap.Int("datapointsRead", datapointsRead))
 	}()
 
 	iter, corruptFiles, err := s.newIteratorFn(iterOpts)
 	if err != nil {
-		return nil, fmt.Errorf("unable to create commit log iterator: %v", err)
+		err = fmt.Errorf("unable to create commit log iterator: %v", err)
+		return bootstrap.NamespaceResults{}, err
 	}
 
 	if len(corruptFiles) > 0 {
-		s.logAndEmitCorruptFiles(corruptFiles, true)
+		s.logAndEmitCorruptFiles(corruptFiles)
 		encounteredCorruptData = true
 	}
 
 	defer iter.Close()
 
-	// Setup the M3TSZ encoding pipeline
-	var (
-		// +1 so we can use the shard number as an index throughout without constantly
-		// remembering to subtract 1 to convert to zero-based indexing
-		numShards        = s.findHighestShard(shardsTimeRanges) + 1
-		numConc          = s.opts.EncodingConcurrency()
-		encoderPool      = blOpts.EncoderPool()
-		workerErrs       = make([]int, numConc)
-		shardDataByShard = s.newShardDataByShard(shardsTimeRanges, numShards)
-	)
-
-	encoderChans := make([]chan encoderArg, numConc)
-	for i := 0; i < numConc; i++ {
-		encoderChans[i] = make(chan encoderArg, encoderChanBufSize)
-	}
-
-	// Spin up numConc background go-routines to handle M3TSZ encoding. This must
+	// Spin up numWorkers background go-routines to handle accumulation. This must
 	// happen before we start reading to prevent infinitely blocking writes to
-	// the encoderChans.
-	wg := &sync.WaitGroup{}
-	for workerNum, encoderChan := range encoderChans {
+	// the worker channels.
+	var wg sync.WaitGroup
+	for _, worker := range workers {
+		worker := worker
 		wg.Add(1)
-		go s.startM3TSZEncodingWorker(
-			ns, runOpts, workerNum, encoderChan, shardDataByShard, encoderPool, workerErrs, blOpts, wg)
+		go func() {
+			s.accumulateWorker(worker)
+			wg.Done()
+		}()
 	}
 
-	// Read / M3TSZ encode all the datapoints in the commit log that we need to read.
+	var (
+		// NB(r): Use pointer type for the namespaces so we don't have to
+		// memcopy the large namespace context struct to the work channel and
+		// can pass by pointer.
+		// For the commit log series map we use by value since it grows
+		// large in size and we want to avoid allocating a struct per series
+		// read and just have a by value struct stored in the map (also makes
+		// reusing memory set aside on a per series level between commit
+		// log files much easier to do).
+		commitLogNamespaces    = make([]*bootstrapNamespace, 0, 4)
+		commitLogSeries        = make(map[seriesMapKey]seriesMapEntry)
```

Comment:
Worth Gennyfying?

Reply:
`seriesMapKey` has no strings or anything of any kind so it's just as many allocations to use a non-genny map here (not required).